api/src/main/java/org/pmiops/workbench/monitoring/MonitoringServiceOpenCensusImpl.java

```diff
@@ -0,0 +1,159 @@
+package org.pmiops.workbench.monitoring;
+
+import io.opencensus.exporter.stats.stackdriver.StackdriverStatsConfiguration;
+import io.opencensus.stats.Measure.MeasureDouble;
+import io.opencensus.stats.Measure.MeasureLong;
+import io.opencensus.stats.MeasureMap;
+import io.opencensus.stats.StatsRecorder;
+import io.opencensus.stats.ViewManager;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.inject.Provider;
+import org.jetbrains.annotations.NotNull;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.monitoring.views.MonitoringViews;
+import org.pmiops.workbench.monitoring.views.OpenCensusStatsViewInfo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MonitoringServiceOpenCensusImpl implements MonitoringService {
+  private static final Logger logger =
+      Logger.getLogger(MonitoringServiceOpenCensusImpl.class.getName());
+  private static final String STACKDRIVER_CUSTOM_METRICS_DOMAIN_NAME = "custom.googleapis.com";
+  private ViewManager viewManager;
+  private StatsRecorder statsRecorder;
+  private StackdriverStatsExporterInitializationService
+      stackdriverStatsExporterInitializationService;
+  private Provider<WorkbenchConfig> workbenchConfigProvider;
+
+  @Autowired
+  MonitoringServiceOpenCensusImpl(
+      ViewManager viewManager,
+      StatsRecorder statsRecorder,
+      StackdriverStatsExporterInitializationService stackdriverStatsExporterInitializationService,
+      Provider<WorkbenchConfig> workbenchConfigProvider) {
+    this.viewManager = viewManager;
+    this.statsRecorder = statsRecorder;
+    this.stackdriverStatsExporterInitializationService =
+        stackdriverStatsExporterInitializationService;
+    this.workbenchConfigProvider = workbenchConfigProvider;
+    registerSignals();
+  }
+
+  private void initStatsConfigurationIdempotent() {
+    stackdriverStatsExporterInitializationService.createAndRegister(
+        StackdriverStatsConfiguration.builder()
+            .setMetricNamePrefix(buildMetricNamePrefix())
+            .setProjectId(workbenchConfigProvider.get().server.projectId)
+            .build());
+  }
+
+  private String buildMetricNamePrefix() {
+    return String.format(
+        "%s/%s/",
+        STACKDRIVER_CUSTOM_METRICS_DOMAIN_NAME,
+        workbenchConfigProvider.get().server.shortName.toLowerCase());
+  }
+
+  private void registerSignals() {
+    Arrays.stream(MonitoringViews.values())
+        .map(MonitoringViews::toStatsView)
+        .forEach(viewManager::registerView);
+  }
+
+  @Override
+  public void recordValue(OpenCensusStatsViewInfo viewProperties, Number value) {
+    try {
+      initStatsConfigurationIdempotent();
```

Comment:
@ericsong here's the hack I was talking about. I put it in a separate service so we can safely unit test this monitoring service.